evm-ds/src/main.rs

```diff
@@ -268,63 +172,79 @@ async fn run_evm_impl(
         }));
         // Scale back remaining gas to Scilla units (no rounding!).
         let remaining_gas = executor.gas() / gas_scaling_factor;
-        match result {
+        let result = match result {
             Ok(exit_reason) => {
+                match exit_reason {
```

Comment:
FWIW we could probably flatten these `match`es into a single one by e.g. matching on `Ok(evm::ExitReason::Succeed(_)`, etc.

Reply:
Good point, but  it would de-emphasize the nesting logic of these cases - why do we need that?